shopping_app/lib/screens/home_screen/api_home_screen.dart

```diff
@@ -0,0 +1,183 @@
+import 'dart:convert';
+import 'package:flutter/material.dart';
+import 'package:http/http.dart' as http;
+import 'package:provider/provider.dart';
+import 'package:shopping_app/items/search_by_name_list_widget.dart';
+import 'package:shopping_app/providers/super_hero_provider.dart';
+
+class ApiHomeScreen extends StatefulWidget {
+  const ApiHomeScreen({Key key}) : super(key: key);
+
+  @override
+  _ApiHomeScreenState createState() => _ApiHomeScreenState();
+}
+
+class _ApiHomeScreenState extends State<ApiHomeScreen> {
+  TextEditingController _controller = TextEditingController();
+  bool isLoading = false;
+  bool isInit = true;
+  bool isSearchingByName = false;
+  Map responseMap;
+
+  Future fetchData(int id) async {
+    setState(() {
+      isLoading = true;
+      isSearchingByName = false;
+    });
+    http.Response response;
+    response = await http
```

Comment:
Can we use MVVM architecture ( model and view )

Reply:
Providers are already working as MV **VM** <Change Notifier Provider( Notifying Listeners as View Model ) > architecture. As I think we are already attaching observers to the classes and widgets. As Data Changes ( Acting as LiveData ) this will ensure data to be changed on Every Widget Observing the Data.

View Components  are Already Divided in Widgets for VIEWS
and MODELS are Created for a Single Product / Superhero.

Reply:
View contains only UI part and view model will cover all logics